Jobba.Store.EF/DbContexts/EfJsonComparer.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using Newtonsoft.Json;
+
+namespace Jobba.Store.EF.DbContexts;
+
+public class EfJsonComparer<T> : ValueComparer<T>
```

Comment:
Yes because the type is an interface on those properties which is not supported by EF, only concrete classes. I had to make this work by storing the type with the serialization